### PR TITLE
Add a PlatformViewRenderTarget abstraction

### PR DESCRIPTION
**Pull request by johnmccutchan**
- Introduce PlatformViewRenderTarget interface.
- Refactor VirtualDisplayController and PlatformViewWrapper to extract SurfaceTexturePlatformViewRenderTarget into a separate class.

In a future CL I will add an ImageReaderPlatformViewRenderTarget which will enable Platform Views on Impeller/Vulkan.

Tracking issue: https://github.com/flutter/flutter/issues/130892

### CODE REVIEW

**Review by reidbaker — APPROVED** [review]

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  // NOTE: If this returns null the RenderTarget is "full" and has no room for a
+  // new frame.
+  Canvas lockHardwareCanvas();
+
+  // Forwards call to Surface returned by getSurface.
+  // NOTE: Must be called if lockHardwareCanvas returns a non-null Canvas.
+  void unlockCanvasAndPost(Canvas canvas);
+
+  // The id of this render target.
+  public long id();
```
Comment (stuartmorgan) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java:35]: `getId`?

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  // Release backing resources.
```
Comment (stuartmorgan) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java:37]: Nit: Releases

(For consistency with all the other comments.)

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java [hunk 3]
```diff
@@ -132,61 +76,7 @@ public void setTouchProcessor(@Nullable AndroidTouchProcessor newTouchProcessor)
...
-  protected Surface createSurface(@NonNull SurfaceTexture tx) {
-    return new Surface(tx);
-  }
-
-  /** Returns the texture where the view is projected. */
-  @Nullable
-  public SurfaceTexture getTexture() {
-    return tx;
-  }
+  public void setRenderTarget(@Nullable PlatformViewRenderTarget renderTarget) {}
```
Comment (stuartmorgan) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:79]: Why is this a no-op? Shouldn't it change `this.renderTarget`?

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java (hunk 2 above)
Comment (johnmccutchan) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java:37]: Done.

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java (hunk 3 above)
Comment (johnmccutchan) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:79]: This was dead code. Removed.

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java (hunk 1 above)
Comment (stuartmorgan) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java:35]: The Google Java style guide doesn't explicitly say, but many of the examples there start with `get`, and my impression (maybe incorrect, I don't have a lot of Java background) is that Java convention in general was to use `get` for getters.

Comment (johnmccutchan) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewRenderTarget.java:35]: Done.

**Pull request merged**